app/notifications/validators.py

```diff
@@ -147,17 +148,28 @@ def check_notification_content_is_not_empty(template_with_content):
         raise BadRequestError(message=message)
 
 
-def validate_template(template_id, personalisation, service, notification_type):
+def get_template_dict(template_id, service_id):
+    from app.schemas import template_schema
     try:
-        template = templates_dao.dao_get_template_by_id_and_service_id(
+        fetched_template = templates_dao.dao_get_template_by_id_and_service_id(
             template_id=template_id,
-            service_id=service.id
+            service_id=service_id
         )
     except NoResultFound:
         message = 'Template not found'
         raise BadRequestError(message=message,
                               fields=[{'template': message}])
 
+    return template_schema.dump(fetched_template).data
+
+
+def get_template_model(template_id, service_id):
```

Comment:
We can add a short-lived in memory cache here, for example:
```suggestion
@cachetools.cached(cache=cachetools.TTLCache(ttl=5))
def get_template_model(template_id, service_id):
```

Reply:
Just to clarify, the risk here is that someone was sending notifications out, made a change to their template and then sent another notification within 5 seconds then it would use the old template? I think if we swapped to something like 2 seconds the risk of a user being quick enough to hit that edge case would be lower but I'm not opposed to keeping it at 5.

This would be good for GOV.UK though, reducing say 1000 DB calls to get templates over 5 seconds down to 1 DB call.

Reply:
Yeah, exactly. Can talk about the exact timings when we add the caching, but basing it on how quickly a user could edit a template then send themselves a message would be good.